clar2wasm/tests/wasm-generation/map.rs

```diff
@@ -0,0 +1,122 @@
+use clar2wasm::tools::{crosscheck, crosscheck_compare_only};
+use clarity::vm::types::{ListTypeData, SequenceSubtype, StringSubtype, TypeSignature};
+use proptest::strategy::{Just, Strategy};
+use proptest::{prop_oneof, proptest};
+
+use crate::{list, uint, PropValue, TypePrinter, Value};
+
+fn strategies_for_list() -> impl Strategy<Value = TypeSignature> {
+    prop_oneof![
+        Just(TypeSignature::IntType),
+        Just(TypeSignature::UIntType),
+        (0u32..128).prop_map(|s| TypeSignature::SequenceType(SequenceSubtype::BufferType(
+            s.try_into().unwrap()
+        ))),
+        (0u32..128).prop_map(|s| TypeSignature::SequenceType(SequenceSubtype::StringType(
+            StringSubtype::ASCII(s.try_into().unwrap())
+        )))
+    ]
+}
+
+fn generate_list(ty: TypeSignature) -> impl Strategy<Value = Value> {
+    (8u32..32)
+        .prop_map(move |s| ListTypeData::new_list(ty.clone(), s).unwrap())
+        .prop_flat_map(list)
+}
+
+proptest! {
+    #![proptest_config(super::runtime_config())]
+
+    #[test]
+    fn crossprop_map_lte(
+        vals in strategies_for_list()
+        .prop_map(generate_list)
+        .prop_flat_map(|l| proptest::collection::vec(l, 1..=10))) {
+
+        for chunk in vals.chunks(2) {
+            if let [arg1, arg2] = chunk {
+                let cmd = format!(
+                    "(map <= {} {})",
+                    PropValue(arg1.clone()),
+                    PropValue(arg2.clone())
+                );
+
+                crosscheck_compare_only(&cmd);
```

Comment:
Can't we use `crosscheck` directly here? I have the impression that the result should always be `Ok(Some(Value::Bool(arg1 <= arg2)))`.

Reply:
In simple tests like those I tend to use the compiler and interpreter comparison only. In the end they have the same effect but with `crosscheck_compare_only` the intention to compare only both outputs is more clear.